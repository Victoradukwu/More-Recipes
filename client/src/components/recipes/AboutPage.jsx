import React from 'react';

const AboutPage = () => (
    <div>
    <h4>This a test About page</h4>
    <p> You are welcome to my About page bla bla bla ...</p>
    </div>
  );
export default AboutPage;