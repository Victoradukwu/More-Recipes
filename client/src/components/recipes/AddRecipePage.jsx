import React, { Component } from 'react';
import { connect } from 'react-redux';

class AddRecipe extends Component {
  constructor(props){
    super(props);
  }
  render() {
    console.log(this.props) 
    return (
      <div className="card mx-auto text-center w-50">
        <div className="card-header" id = "cardHeader">
          <h4>Add a recipe</h4>
        </div>
    
        <div className="card-body">
          <div className = "forminfo" style = {{color: 'blue'}}>
            To submit a recipe to the catalog, fill out the form below and submit.
          </div>
          <br/><br/>
          <form action="">                    
            <fieldset className="form-group">
              <label htmlFor="name">Name of recipe:</label>
              <input type="text" className="form-control form-input" id="name" name="name"/>
            </fieldset>
            <fieldset className="form-group">
              <label htmlFor="name">Category:</label>
              <input type="text" className="form-control form-input" id="Category" name="Category"/>
            </fieldset>
            <fieldset className="form-group">
              <label htmlFor="ingredients">Ingredients:</label><br/>>
              <textarea nameName="ingredients" id="ingredients" cols="30" rows="7" className="form-control form-input" />
            </fieldset>                    
            <fieldset className="form-group">
              <label htmlFor="steps">Instructions:</label><br/>>
              <textarea nameName="instructions" id="instructions" cols="30" rows="7" className="form-control form-input" />
            </fieldset>                    =
            <button type="submit" className="btn" style = {{ backgroundColor: '#336600', color: 'white' }}>Submit Recipe</button>
          </form>
        </div>
      </div>
        );
  };
}


export default AddRecipe;