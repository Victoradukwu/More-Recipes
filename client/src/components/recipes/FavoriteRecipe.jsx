import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

const FavoriteRecipe = ({ userFavorite }) => (
  <tr>
    <td> {userFavorite.Recipe.recipeName}</td>
    <td>{userFavorite.category}</td>
    <td>
      <Link
        to={`/recipeDetail/${userFavorite.Recipe.id}`}
        className="btn search"
      >Check Details
      </Link>
    </td>

  </tr>
);

FavoriteRecipe.propTypes = {
  userFavorite: PropTypes.object.isRequired
};

export default FavoriteRecipe;
