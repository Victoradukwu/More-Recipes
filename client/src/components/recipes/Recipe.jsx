import { Link } from 'react-router-dom';
import React from 'react';
import PropTypes from 'prop-types';

const Recipe = ({ recipe }) => (
  <div className="card-holder">
    <div className="card bg-light text-black">
      <img
        className="card-img"
        src={recipe.recipePicture || 'http://res.cloudinary.com/victoradukwu/image/upload/v1520512629/hd1.jpg'}
        alt="Sample recipe"
        style={{ height: '250px' }}
      />
      <p
        className="card-text mt-0"
        style={{ backgroundColor: 'white', padding: '10px' }}
      >
        <span
          className="fa fa-thumbs-up fa-lg "
          style={{ float: 'left' }}
        >{recipe.upvote}
        </span>
        <span
          className="fa fa-thumbs-down fa-lg"
          style={{ float: 'right' }}
        >{recipe.downvote}
        </span>
      </p>

      <div className="card-body">
        <h4 className="card-title">{recipe.recipeName}</h4>

        <Link
          to={`/recipeDetail/${recipe.id}`}
          className="btn search"
        >Check Details
        </Link>
      </div>
    </div>
  </div>
);

Recipe.propTypes = {
  recipe: PropTypes.shape({
    id: PropTypes.number,
    recipeName: PropTypes.string,
    category: PropTypes.string,
    ingredients: PropTypes.string,
    instructions: PropTypes.string,
    upvote: PropTypes.number,
    downvote: PropTypes.number,
    views: PropTypes.number,
    favorites: PropTypes.number,
    createdAt: PropTypes.any,
    updatedAt: PropTypes.any,
    userId: PropTypes.number,
    reviews: PropTypes.array,
    User: PropTypes.shape({ id: PropTypes.number, name: PropTypes.string })
  }).isRequired,
};
Recipe.defaultProps = {

};

export default Recipe;
