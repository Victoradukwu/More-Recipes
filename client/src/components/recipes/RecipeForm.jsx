import React from 'react';
import PropTypes from 'prop-types';
import TextInput from '../common/TextInput';
// import ImageInput from '../common/ImageInput';

const RecipeForm = ({ recipe, onChange, errors, onSubmit }) => (
  <form onSubmit={onSubmit}>
    <TextInput
      name="recipeName"
      label="Recipe Name"
      value={recipe.recipeName}
      onChange={onChange}
      error={errors.recipeName}
    />
    <TextInput
      name="category"
      label="Category"
      value={recipe.category}
      onChange={onChange}
      error={errors.caregory}
    />
    <TextInput
      name="ingredients"
      label="Ingredients"
      value={recipe.ingredients}
      onChange={onChange}
      error={errors.ingredients}
    />
    <TextInput
      name="instruction"
      label="Instruction"
      value={recipe.instruction}
      onChange={onChange}
      error={errors.instruction}
    />
    {/* <ImageInput
      name="recipePicture"
      label="RecipePicture"
      value={recipe.recipePicture}
      onChange={onChange}
      error={errors.recipePicture}
    /> */}
    <button
      type="submit"
      className="btn"
      style={{ backgroundColor: '#336600', color: 'white' }}
    >
      Submit Recipe
    </button>


  </form>
);

RecipeForm.propTypes = {
  recipe: PropTypes.object,
  onChange: PropTypes.func.isRequired,
  errors: PropTypes.objectOf(PropTypes.string),
  onSubmit: PropTypes.func.isRequired
};

RecipeForm.defaultProps = {
  recipe: {},
  errors: {}
};

export default RecipeForm;
