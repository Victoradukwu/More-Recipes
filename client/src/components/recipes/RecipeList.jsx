import React from 'react';
import PropTypes from 'prop-types';
import UserRecipe from './UserRecipe';

const RecipeList = ({ userRecipes }) => {
  return (
    <table className="table">
      <thead>
        <tr>
          <th>Recipe Name</th>
          <th>Upvotes</th>
          <th>Downvotes</th>
          <th>Favorites</th>
          <th>views</th>
          <th>Actions</th>        
        </tr>
      </thead>
      <tbody>
        { userRecipes.map(userRecipe =>
          <UserRecipe key={userRecipe.Id} userRecipe={userRecipe} />) }

      </tbody>
    </table>
  );
};
RecipeList.propTypes = {
  userRecipes: PropTypes.array.isRequired
};

export default RecipeList;
