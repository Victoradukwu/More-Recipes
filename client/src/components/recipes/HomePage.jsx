import React, { Component } from 'react';
import { connect } from 'react-redux';
import ReactPaginate from 'react-paginate';
import PropTypes from 'prop-types';

import Recipe from './Recipe';

import { fetchRecipes } from '../../actions/recipeActions';

class HomePage extends Component {
  constructor(props) {
    super(props);
    this.handlePageClick = this.handlePageClick.bind(this);

    this.state = { recipes: [] };
  }

  componentDidMount() {
    this.props.fetchRecipes('/api/v1/recipes?sort=upvotes&order=des');
  }

  handlePageClick(data) {
    const selected = data.selected + 1;
    this.props.fetchRecipes(`/api/v1/recipes?sort=upvotes&order=des&page=${selected}`);
  }

  render() {
    return (
      <div>
        <div className="container-fluid">
          <img alt="" src={require('../../assets/img/img001.jpeg')} style={{ width: '100%', height: '620px' }} />
        </div>
        <h4>
          <small>Our top picks</small>
        </h4>
        <hr />
        <div className="container">
          <div className="container bg-3 text-center main-content">
            <div className="grid-holder">
              {this.props.payload.recipes.map(recipe => (
                <Recipe key={recipe.id} recipe={recipe} />
          ))}
            </div>
          </div>
          <br />

          <ReactPaginate
            previousLabel="&laquo;"
            nextLabel="&raquo;"
            breakClassName="break-me"
            pageCount={this.props.payload.pages}
            marginPagesDisplayed={2}
            pageRangeDisplayed={5}
            onPageChange={this.handlePageClick}
            containerClassName="pagination pagination-lg justify-content-center"
            subContainerClassName="page-item"
            activeClassName="active"
          />
          <br /><br />
          <br /><br />
        </div>
      </div>
    );
  }
}
HomePage.propTypes = {
  fetchRecipes: PropTypes.func.isRequired
};

const mapStateToProps = state => ({
  isloading: state.fetchRecipeRequest,
  hasErrored: state.fetchRecipeFailure,
  payload: state.fetchRecipeSuccess
});

const mapDispatchToProps = dispatch => ({
  fetchRecipes: url => dispatch(fetchRecipes(url))
});
export default connect(mapStateToProps, mapDispatchToProps)(HomePage);
