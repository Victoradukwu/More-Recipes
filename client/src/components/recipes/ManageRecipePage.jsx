import React, { Component } from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import PropTypes from 'prop-types';
import toastr from 'toastr';
import axios from 'axios';
import * as recipeActions from '../../actions/recipeActions';
import RecipeForm from './RecipeForm';
import validate from '../../helpers/validate';
import checkImageFile from '../../helpers/checkImageFile';
import setAuthorizationToken from '../../helpers/setAuthorizationToken';

require('dotenv').config();

/**
 * @description
 * @class ManageRecipePage
 * @extends {Component}
 */
class ManageRecipePage extends Component {
  constructor(props) {
    super(props);
    this.state = {
      recipe: Object.assign({}, props.recipe),
      errors: {},
      defaultImgSrc: '../../assets/img/hd8.jpg',
    };

    this.onChange = this.onChange.bind(this);
    this.onSubmit = this.onSubmit.bind(this);
    this.handleImageChange = this.handleImageChange.bind(this);
  }

  componentWillReceiveProps(nextProps) {
    if (nextProps.error.status) {
      this.setState({ errors: nextProps.error });
    }
    if (this.props.recipe.id !== nextProps.recipe.id) {
      this.setState({ recipe: Object.assign({}, nextProps.recipe) });
    }
  }
  onChange(event) {
    const field = event.target.name;
    const myRecipe = Object.assign({}, this.state.recipe);
    myRecipe[field] = event.target.value;
    return this.setState({ recipe: myRecipe });
  }

  async onSubmit(event) {
    event.preventDefault();
    const recipeObject = {
      recipeName: event.target.recipeName.value,
      category: event.target.category.value,
      ingredients: event.target.ingredients.value,
      instructions: event.target.instructions.value,
      id: this.state.recipe.id
    };
    if (this.state.recipe.recipePicture) {
      const recipeImage = await this.getImgURL();
      const imageUrl = recipeImage.data.url;
      recipeObject.recipePicture = imageUrl;
    }
    this.setState({ errors: {} });

    setAuthorizationToken(localStorage.getItem('token'));
    const { isValid, errors } = validate(recipeObject);
    if (isValid) {
      this.setState({ errors: {} });
      this.props.actions.submitRecipe(recipeObject)
        .then(() => {
          this.props.history.push('/myRecipes');
        });
    }
    this.setState({ errors });
  }

  getImgURL() {
    const imgPix = this.state.recipe.recipePicture;
    delete axios.defaults.headers.common['x-access-token'];
    const imageData = new FormData();
    imageData.append('file', imgPix);
    imageData.append('upload_preset', process.env.CLOUDINARY_UPLOAD_PRESET);
    return axios.post(process.env.CLOUDINARY_URL, imageData);
  }

  handleImageChange(event) {
    event.persist();
    if (event.target.files && event.target.files[0]) {
      const file = event.target.files[0];
      const filereader = new FileReader();
      checkImageFile(filereader, file, (fileType) => {
        if (fileType === 'image/png' || fileType === 'image/gif' ||
          fileType === 'image/jpeg') {
          this.setState({
            recipe:
            Object.assign({}, this.state.recipe, { recipePicture: file })
          });

          filereader.onload = (e) => {
            this.setState({ defaultImgSrc: e.target.result });
          };
          filereader.readAsDataURL(file);
        } else {
          toastr.error('image must be in png, jpeg or gif format');
        }
      });
    } else {
      this.setState({ defaultImgSrc: '../../assets/img/hd8.jpg' });
    }
  }

  isSignedIn() {
    if (localStorage.token === undefined) {
      this.props.history.push('/signin');
    }
  }
  render() {
    this.isSignedIn();
    return (
      <div className="row">
        <div className="col-sm-3" />
        <div className="col-sm-6">
          <h3> Add/Edit Recipe</h3>
          <RecipeForm
            recipe={this.state.recipe}
            errors={this.state.errors}
            onSubmit={this.onSubmit}
            onChange={this.onChange}
            onImageUpload={this.handleImageChange}
            defaultImgSrc={this.state.defaultImgSrc}
          />
        </div>
        <div className="col-sm-3" />
      </div>
    );
  }
}
ManageRecipePage.propTypes = {
  error: PropTypes.object,
  history: PropTypes.shape({
    push: PropTypes.func
  }).isRequired,
  actions: PropTypes.object.isRequired,
  recipe: PropTypes.shape({
    id: PropTypes.number,
    recipeName: PropTypes.string,
    category: PropTypes.string,
    ingredients: PropTypes.string,
    instructions: PropTypes.string,
    upvote: PropTypes.number,
    downvote: PropTypes.number,
    views: PropTypes.number,
    favorites: PropTypes.number,
    createdAt: PropTypes.any,
    updatedAt: PropTypes.any,
    userId: PropTypes.number,
    reviews: PropTypes.array,
    User: PropTypes.shape({ id: PropTypes.number, name: PropTypes.string })
  }).isRequired,
};

ManageRecipePage.defaultProps = {
  error: {}
};
const getRecipeById = (recipes, id) => {
  const singleRecipe = recipes.filter(recipe =>
    parseInt(recipe.id, 10) === parseInt(id, 10));
  if (singleRecipe.length > 0) {
    return singleRecipe[0];
  }
  return 'Recipe Not Found';
};
const mapStateToProps = (state, ownProps) => {
  const recipeId = ownProps.match.params.id;
  let recipe = { recipeName: '', instructions: '', category: '' };
  if (recipeId && state.userRecipes.length > 0) {
    recipe = getRecipeById(state.userRecipes, recipeId);
  }
  return { recipe };
};
const mapDispatchToProps = dispatch => ({
  actions: bindActionCreators(recipeActions, dispatch)
});

export default connect(mapStateToProps, mapDispatchToProps)(ManageRecipePage);
