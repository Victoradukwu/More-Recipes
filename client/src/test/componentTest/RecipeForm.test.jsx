import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import { RecipeForm } from '../../components/recipes/RecipeForm';

describe('Recipe Form', () => {
  const props = {
    onChange: jest.fn(),
    onImageUpload: jest.fn(),
    errors: {},
    onSubmit: jest.fn(),
    defaultImgSrc: '',
  };

  it('Should render RecipeForm Correctly', () => {
    const wrapper = shallow(<RecipeForm {...props} />);
    expect(wrapper.find('form').exists()).toBeTruthy();
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
