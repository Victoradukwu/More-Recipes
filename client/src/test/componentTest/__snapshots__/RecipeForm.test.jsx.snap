// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Recipe Form Should render RecipeForm Correctly 1`] = `
<form
  onSubmit={[MockFunction]}
>
  <TextInput
    error=""
    label="Recipe Name"
    name="recipeName"
    onChange={[MockFunction]}
    placeholder=""
    value=""
  />
  <TextInput
    error=""
    label="Category"
    name="category"
    onChange={[MockFunction]}
    placeholder=""
    value=""
  />
  <TextArea
    error=""
    label="Ingredients"
    name="ingredients"
    onChange={[MockFunction]}
    placeholder=""
    value=""
  />
  <TextArea
    error=""
    label="Instruction"
    name="instructions"
    onChange={[MockFunction]}
    placeholder=""
    value=""
  />
  <div
    className="row"
  >
    <div
      className="col-sm-6"
    >
      <ImageInput
        error=""
        label="Recipe Picture"
        name="recipePicture"
        onChange={[MockFunction]}
        placeholder=""
        value=""
      />
    </div>
    <div
      className="col-sm-6"
    >
      <img
        alt="recipe img"
        className="img-thumbnail img-fluid mb-3"
        src=""
        style={
          Object {
            "height": "150px",
            "width": "150px",
          }
        }
      />
    </div>
  </div>
  <button
    className="btn btn-lg"
    style={
      Object {
        "backgroundColor": "#336600",
        "color": "white",
      }
    }
    type="submit"
  >
    Submit Recipe
  </button>
</form>
`;
